### ISSUE TITLE
[BUG] Extra * added in xml files

### ISSUE THREAD

**Issue by starbucks2010** [iBotPeaches/Apktool#2771]
### Information
1. **Apktool Version (`apktool -version`)** - 2.6.1
2. **Operating System (Mac, Linux, Windows)** - Windows
3. **APK From? (Playstore, ROM, Other)** - ROM

### Stacktrace/Logcat

```
Include stacktrace here
```

### Steps to Reproduce
1. `apktool ` Just decode the SystemUI.apk and look into the xml files for the extra * added on code pointing to framework resources like this: android:textSize="@*android:dimen/text_size_.....

apktool v2.6.0 : android:textSize="@android:dimen/text_size_body_2_material"
apktool v2.6.1: android:textSize="@*android:dimen/text_size_body_2_material"

### Frameworks
If this APK is from an OEM ROM (Samsung, HTC, LG). Please attach framework files
(`.apks` that live in `/system/framework` or `/system/priv-app`)

Framework and SystemUI uploaded here: https://drive.google.com/drive/folders/114MkIArkcbQeDqO702RTI-e3hllxJtOd?usp=sharing

### APK
If this APK can be freely shared, please upload/attach a link to it.

### Questions to ask before submission
1. Have you tried `apktool d`, `apktool b` without changing anything? - Yes, same issue
3. If you are trying to install a modified apk, did you resign it? - Just decompiling
4. Are you using the latest apktool version? - Yes, v2.6.1

This issue seems to be related to bug reference #2650.

**Comment by iBotPeaches** [comment]
So I took a look in detail on this. I attached a breakpoint and waited for the first resource to be detected as private.

It was

```
0x7f1403bd style/Theme.SystemUI
```

Which is a private resource specific to that application. So I'm confused why this is attaching a private resource trick when I thought that was isolated to framework references.

@MrIkso - Should the #2650 fix been applied only to non-private packageIds? Like we we ignore any on 7F, since its presumed those resources are private and internal. Or am I misunderstanding the patch?

**Comment by pashamcr** [comment]
@iBotPeaches delete commit https://github.com/iBotPeaches/Apktool/commit/d29411117e3b2da726cfd38ef418bd5b7a18ddb7
compile new aapt/aapt2 from current sources, this problem will disappear by itself, besides this fixes the error https://github.com/iBotPeaches/Apktool/issues/2033 (apk https://apkpure.com/contacts/com.google.android.contacts/variant/3.62.2.428100364-APK)

For example, aapt2_x64 from new sources, embed it in apktool (just an archiver) and check:

[aapt2.exe.txt](https://github.com/iBotPeaches/Apktool/files/8205903/aapt2.exe.txt)

**Comment by iBotPeaches** [comment]
Thanks @pashamcr, do you have a commit off hand for aapt/aapt2 that resolved this? What led aapt to handling private resources less strict than present?

**Comment by pashamcr** [comment]
I don’t remember what I fixed, but there seems to be all the fixes like yours
```
cd frameworks
patch -p0 < aapt2.patch
```
[aapt2.patch.txt](https://github.com/iBotPeaches/Apktool/files/8207753/aapt2.patch.txt)

**Comment by iBotPeaches** [comment]
Okay thanks, looks like I'll investigate this and look into a revert and updating aapt2.

**Comment by iBotPeaches** [comment]
```
➜  2771 apktool d SystemUI_g998b_bvb1_stock.apk -t 2771
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on -Dswing.aatext=true
I: Using Apktool 2.6.2-6c94ca-SNAPSHOT on SystemUI_g998b_bvb1_stock.apk
I: Loading resource table...
I: Decoding AndroidManifest.xml with resources...
I: Loading resource table from file: /home/ibotpeaches/.local/share/apktool/framework/1-2771.apk
I: Regular manifest package...
I: Decoding file-resources...
I: Decoding values */* XMLs...
I: Baksmaling classes.dex...
I: Baksmaling classes2.dex...
I: Baksmaling classes3.dex...
I: Copying assets and libs...
I: Copying unknown files...
I: Copying original files...
I: Copying META-INF/services directory
➜  2771
```

Checked for `*`.

```
➜  SystemUI_g998b_bvb1_stock grep -ir 'text_size_body_2_material' *
res/values/public.xml:    <public type="dimen" name="abc_text_size_body_2_material" id="0x7f07002b" />
res/values/styles.xml:        <item name="android:textSize">@dimen/abc_text_size_body_2_material</item>
res/values/dimens.xml:    <dimen name="abc_text_size_body_2_material">14.0sp</dimen>
```

Built
```
➜  SystemUI_g998b_bvb1_stock apktool b . --use-aapt2
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on -Dswing.aatext=true
I: Using Apktool 2.6.2-6c94ca-SNAPSHOT
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether resources has changed...
I: Building resources...
I: Copying libs... (/kotlin)
I: Copying libs... (/META-INF/services)
I: Building apk file...
I: Copying unknown files/dir...
I: Built apk...
➜  SystemUI_g998b_bvb1_stock 
```

Will be resolved with #2787
